modules/gapi/samples/onevpl_infer_single_roi.cpp

```diff
@@ -365,40 +495,102 @@ int main(int argc, char *argv[]) {
+#ifdef __linux__
+#if defined(HAVE_VA) || defined(HAVE_VA_INTEL)
+        static const char *predefined_vaapi_devices_list[] {"/dev/dri/renderD128",
+                                                            "/dev/dri/renderD129",
+                                                            "/dev/dri/card0",
+                                                            "/dev/dri/card1",
+                                                            nullptr};
+        std::stringstream ss;
+        int device_fd = -1;
+        VADisplay va_handle = nullptr;
+        for (const char **device_path = predefined_vaapi_devices_list;
+            *device_path != nullptr; device_path++) {
+            device_fd = open(*device_path, O_RDWR);
+            if (device_fd < 0) {
+                std::string info("Cannot open GPU file: \"");
+                info = info + *device_path + "\", error: " + strerror(errno);
+                ss << info << std::endl;
+                continue;
+            }
+            va_handle = vaGetDisplayDRM(device_fd);
+            if (!va_handle) {
+                close(device_fd);
+                std::string info("VAAPI device vaGetDisplayDRM failed, error: ");
+                info += strerror(errno);
+                ss << info << std::endl;
+                continue;
+            }
+            int major_version = 0, minor_version = 0;
+            VAStatus status {};
+            status = vaInitialize(va_handle, &major_version, &minor_version);
+            if (VA_STATUS_SUCCESS != status) {
+                close(device_fd);
+                va_handle = nullptr;
+
+                std::string info("Cannot initialize VAAPI device, error: ");
+                info += vaErrorStr(status);
+                ss << info << std::endl;
+                continue;
+            }
+            std::cout << "VAAPI created for device: " << *device_path << ", version: "
+                      << major_version << "." << minor_version << std::endl;
+            break;
+        }
 
-        // put accel type description for VPL source
-        source_cfgs.push_back(cfg::create_from_string(
-                                        "mfxImplDescription.AccelerationMode"
-                                        ":"
-                                        "MFX_ACCEL_MODE_VIA_D3D11"));
+        // check device creation
+        if (!va_handle) {
+            std::cerr << "Cannot create VAAPI device. Log:\n" << ss.str() << std::endl;
+            return -1;
+        }
+        gpu_accel_device = cv::util::make_optional(
+                            cv::gapi::wip::onevpl::create_vaapi_device(reinterpret_cast<void*>(va_handle),
+                                                                       "GPU", device_fd));
+        gpu_accel_ctx = cv::util::make_optional(
+                            cv::gapi::wip::onevpl::create_vaapi_context(nullptr));
+#endif // defined(HAVE_VA) || defined(HAVE_VA_INTEL)
+#endif // #ifdef __linux__
```

Comment:
We should absorb all this complexity someday. User apps shouldn't be *that* complicated to use a VA path..

Reply:
I put comment above:
https://github.com/opencv/opencv/pull/22212/files#diff-b4563bcdbe86eb5c8f1f2cce227809b02980777b25f5f4be2bb717826bedbebbR414


The idead behind the current sample is demonstration of the most complex way how to use VPL with GPU. This complexity represent a way HOW-TO-CHOOSE-SUITABLE-GPU from multi-acceleration hardware setup. When you have two or more GPUs then it should provide stable and clearest way how to enumerate them and use as pipeline accelerator.

Additionally, user might have their own framework/pipeline which uses PREdefined GPU device from own framework primitives AND i do not see reasons why we should not REuse this straight device handle with native API instead of  using VPL/GAPI "default" one (simplest but not worth way). IN more complicated case use-predefined GPU might be differen when VPL/GAPI/IE recreate it by itself by using "default"  selection.  S0, for REAL scenarious (not synthetic one) we must implement accepting user-defiend device like these


So, I had intention to explain the most customizable scenarion by this sample (which is trickly and maybe complicated for newcomers), because the default GPU device selection is pretty simple: just invoke default onevpl::GSource ctor with like this:

```
auto s = onevpl::GSource( CfgParam {"mfxImplDescription.AccelerationMode","MFX_ACCEL_MODE_VIA_D3D11"});
```